impl/src/main/java/impl/AnswerResourceImpl.java

```diff
@@ -30,29 +38,76 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(AnswerResourceImpl.class);
 
-    private final AnswerDao       answerDao;
-    private final QuestionDao     questionDao;
-    private final DaoTransactions daoTransactions;
+    private final AnswerDao         answerDao;
+    private final QuestionDao       questionDao;
+    private final DaoTransactions   daoTransactions;
+    private final SlackResource     slackResource;
+    private final UserResource      userResource;
+    private final ApplicationConfig applicationConfig;
 
     @Inject
-    public AnswerResourceImpl(AnswerDao answerDao, QuestionDao questionDao, DaoTransactions daoTransactions) {
+    public AnswerResourceImpl(AnswerDao answerDao,
+        QuestionDao questionDao,
+        DaoTransactions daoTransactions,
+        SlackResource slackResource,
+        UserResource userResource,
+        ApplicationConfig applicationConfig
+    ) {
         this.answerDao = answerDao;
         this.questionDao = questionDao;
         this.daoTransactions = daoTransactions;
+        this.slackResource = slackResource;
+        this.userResource = userResource;
+        this.applicationConfig = applicationConfig;
     }
 
     @Override
     public Observable<Answer> answerQuestion(Auth auth, Answer answer, long questionId) {
         Objects.requireNonNull(answer.getAnswer());
 
         return this.answerDao.createAnswer(auth.getUserId(), questionId, answer)
-            .map(generatedKey -> {
+            .flatMap(generatedKey -> {
                 answer.setId(generatedKey.getKey());
-                return answer;
+                return first(notifyQuestionOwner(answer, questionId)).thenReturn(answer);
             }).onErrorResumeNext(throwable ->
                 error(new WebException(INTERNAL_SERVER_ERROR, ERROR_ANSWER_NOT_CREATED, throwable)));
     }
 
+    private Observable<Void> notifyQuestionOwner(Answer answer, long questionId) {
+        return questionDao.getQuestionById(questionId)
+            .flatMap(question ->
+                userResource.getUserById(question.getUserId())
+                    .flatMap(user -> slackResource.getUserId(user.getEmail()))
+                    .flatMap(slackUserId ->
+                        slackResource.postMessageToSlackAsBotUser(slackUserId, notificationMessage(answer, question))))
+            .onErrorResumeNext(throwable -> {
+                LOG.warn("Could not notify question owner", throwable);
```

Comment:
maybe add who the question owner is, to see if its a issue with the accout etc

Reply:
Fixed